### PR TITLE
add open telemetry provider (#36817)

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

fix https://github.com/istio/istio/issues/37199

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [x] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by zirain** [comment]
as https://github.com/istio/istio/pull/36985 is shipped to 1.13, also cherrypick this MR
@istio/release-managers-1-13 ptal.

**Review by howardjohn — APPROVED** [review]

**Comment by zirain** [comment]
as 1.13.0 is released, can we shipped this in 1.13.1? @istio/release-managers-1-13

File: releasenotes/notes/36817.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: telemetry
+releaseNotes:
+- |
+  **Added** Implement OpenTelemetry Access Log.
```
Comment (ericvn) [inline releasenotes/notes/36817.yaml:6]:
```suggestion
  **Added** OpenTelemetry Access Log.
```

**Comment by douglas-reid** [comment]
This was in the release notes for 1.13, so we should get this in asap.

**Comment by zirain** [comment]
ping @istio/release-managers-1-13

**Review by lei-tang — APPROVED** [review]

**Pull request merged**